pkg/operation/etcdencryption/encryptionkeys.go

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package encryptionconfiguration
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	apiserverconfigv1 "k8s.io/apiserver/pkg/apis/config/v1"
+)
+
+func createEncryptionKey() (apiserverconfigv1.Key, error) {
+	var key apiserverconfigv1.Key
+	keyString, err := createKeySecret()
+	if err != nil {
+		return key, err
+	}
+	if len(keyString) == 0 {
+		return key, fmt.Errorf("generated random encryption key is of unexpected length 0")
+	}
+	key.Name = createKeyName()
```

Comment:
instead of defining the `var` upfront, just return `apiserverconfigv1.Key{Name: ...}` directly.
Also, you might consider returning a pointer instead of a struct.